umodbus/modbus.py

```diff
@@ -653,6 +649,18 @@ def _set_single_reg_in_dict(self,
 
         self._register_dict[reg_type][address] = data
 
+    @overload
```

Comment:
could you please explain why these overload functions are necessary?

Reply:
> could you please explain why these overload functions are necessary?

They're useful for IDEs to recognize return types when using different function codes - for example, if you specify that a coil from address X will be removed, VS Code's Pylance will flag an error if you attempt to assign the result to a variable which contains an integer. I should probably fix the `reg_type` hint to be an `int` instead of `str` to match the `Literal` type.